packages/wdio-types/src/Options.ts

```diff
@@ -8,22 +8,17 @@ import type { ReporterEntry } from './Reporters.js'
 
 export type WebDriverLogTypes = 'trace' | 'debug' | 'info' | 'warn' | 'error' | 'silent'
 export type SupportedProtocols = 'webdriver' | 'devtools' | './protocol-stub.js'
-export type Agents = { http?: any, https?: any }
 
 export type Method = 'GET' | 'POST' | 'PUT' | 'PATCH' | 'HEAD' | 'DELETE' | 'OPTIONS' | 'TRACE' | 'get' | 'post' | 'put' | 'patch' | 'head' | 'delete' | 'options' | 'trace'
 
 export interface RequestLibOptions {
-    agent?: Agents
-    followRedirect?: boolean
```

Comment:
Do we need `RequestLibOptions` at this point? We can just rely on the `fetch` types since we use it for all requests. I think we had this type to ensure that requests made through `got` (in Node.js environments) and `ky` (in browser environments) are compatible. We had this to support https://www.npmjs.com/package/web2driver.

Reply:
I guess we might need this. After extending `RequestInit`, below are the additional options which are used internally. Please share your thoughts, 

```typescript
export interface RequestLibOptions extends RequestInit {
    json?: Record<string, unknown>;
    searchParams?: Record<string, string | number | boolean | null | undefined> | URLSearchParams;
    timeout?: number;
    url?: URL;
    path?: string;
    username?: string;
    password?: string;
}
```

Reply:
We should update to the `RequestInit` interface everywhere, e.g.

- `json`: use `body: opts.body ?? JSON.stringify(opts.json)`
- `searchParams`: use `URLSearchParams` instead and add it to the url
- `timeout`: use AbortController as mentioned [here](https://stackoverflow.com/a/50101022)
- `url` remove as it is passed as second argument
- `path` should not be needed anymore, I think it makes sense to have a helper function that can compose the full url based on connection credential options
- `username` and `password`: use `headers.set('Authorization', 'Basic ' + btoa(username + ":" + password))`

What do you think? Can we introduce these further cleanups? I am not sure how many times we pass along `RequestLibOptions` but it would be idea if this is all using cross platform features.

Reply:
Certainly, the cleanup sounds logical. I'll get started on it.

Reply:
As suggested, I've made the cleanup on `RequestLibOptions`. Regarding moving `timeout` to `AbortController`, I have not done because we're using it in the `request.ts` file where we handle timeout clearance. I'm open to your thoughts on this.

Reply:
Can we get rid of it completely?